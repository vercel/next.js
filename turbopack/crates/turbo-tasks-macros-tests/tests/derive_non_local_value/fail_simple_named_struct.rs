#![allow(dead_code)]

use turbo_tasks::NonLocalValue;

struct UnresolvedValue;

#[derive(NonLocalValue)]
struct ContainsUnresolvedValueNamed {
    a: UnresolvedValue,
}

fn main() {}
