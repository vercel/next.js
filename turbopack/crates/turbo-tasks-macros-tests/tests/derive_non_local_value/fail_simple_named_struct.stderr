error[E0277]: the trait bound `UnresolvedValue: NonLocalValue` is not satisfied
 --> tests/derive_non_local_value/fail_simple_named_struct.rs:9:8
  |
9 |     a: UnresolvedValue,
  |        ^^^^^^^^^^^^^^^ the trait `NonLocalValue` is not implemented for `UnresolvedValue`
  |
  = help: the following other types implement trait `NonLocalValue`:
            &T
            &mut T
            ()
            (A, Z, Y, X, W, V, U, T)
            (B, A, Z, Y, X, W, V, U, T)
            (C, B, A, Z, Y, X, W, V, U, T)
            (D, C, B, A, Z, Y, X, W, V, U, T)
            (E, D, C, B, A, Z, Y, X, W, V, U, T)
          and $N others
note: required by a bound in `DeriveNonLocalValueAssertion::assert_impl_NonLocalValue`
 --> tests/derive_non_local_value/fail_simple_named_struct.rs:7:10
  |
7 | #[derive(NonLocalValue)]
  |          ^^^^^^^^^^^^^ required by this bound in `DeriveNonLocalValueAssertion::assert_impl_NonLocalValue`
  = note: this error originates in the derive macro `NonLocalValue` (in Nightly builds, run with -Z macro-backtrace for more info)
